src/snowflake/connector/query_context_cache.py

```diff
@@ -115,23 +117,26 @@ def _replace_qce(
     ) -> None:
         self._remove_qce(old_qce)
         self._add_qce(new_qce)
+        
+    def sync_priority_map(self):
+        """
+        Sync the _intermediate_priority_map with the _priority_map at the end of the current round of merges.
+        """
+        logger.debug(f"syncPriorityMap called priority_map size = {len(self._priority_map)}, new_priority_map size = {len(self._intermediate_priority_map)}")
+        
+        for key, value in self._intermediate_priority_map.items():
+            self._priority_map[key] = value
```

Comment:
nit: You could potentially do `self._priority_map.update(self._intermediate_priority_map)`

Reply:
updated it according to your suggestion